include/seqan3/alignment/pairwise/alignment_selector.hpp

```diff
@@ -0,0 +1,145 @@
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides seqan3::detail::alignment_selector.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <functional>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include <seqan3/alignment/configuration/all.hpp>
+#include <seqan3/alignment/pairwise/align_result.hpp>
+#include <seqan3/alignment/pairwise/edit_distance_unbanded.hpp>
+#include <seqan3/alphabet/gap/gapped.hpp>
+#include <seqan3/core/concept/tuple.hpp>
+#include <seqan3/core/metafunction/range.hpp>
+#include <seqan3/core/type_list.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Helper metafunction to determine the alignment result type based on the configuration.
+ * \ingroup pairwise
+ * \tparam seq1_t          The type of the first sequence.
+ * \tparam seq2_t          The type of the second sequence.
+ * \tparam configuration_t The configuration type. Must be of type seqan3::detail::configuration
+ */
+template <typename seq1_t, typename seq2_t, typename configuration_t>
+struct determine_result_type
+{
+    //!\brief Helper function to determine the actual result type.
+    static constexpr auto _determine()
+    {
+        using seq1_value_type = value_type_t<std::remove_reference_t<seq1_t>>;
+        using seq2_value_type = value_type_t<std::remove_reference_t<seq2_t>>;
+        using score_type      = int32_t;
+
+        if constexpr (has_align_cfg_v<align_cfg::id::output, configuration_t>)
+        {
+            if constexpr (get<align_cfg::id::output>(configuration_t{}) ==
+                               align_result_key::end)
+                return align_result<type_list<uint32_t,
+                                              score_type,
+                                              std::pair<size_t, size_t>>>{};
+            else if constexpr (get<align_cfg::id::output>(configuration_t{}) ==
+                               align_result_key::begin)
+                return align_result<type_list<uint32_t,
+                                              score_type,
+                                              std::pair<size_t, size_t>,
+                                              std::pair<size_t, size_t>>>{};
+            else if constexpr (get<align_cfg::id::output>(configuration_t{}) ==
+                               align_result_key::trace)
+                return align_result<type_list<uint32_t,
+                                              score_type,
+                                              std::pair<size_t, size_t>,
+                                              std::pair<size_t, size_t>,
+                                              std::tuple<std::vector<gapped<seq1_value_type>>,
+                                                         std::vector<gapped<seq2_value_type>>>>>{};
```

Comment:
no, this will be configurable by the user, but we don't have the gapped_sequence yet. So this is going to be changed then. Is just needed something that works now.